src/lib/connectNonCartesianSubplotToLayout.js

```diff
@@ -23,32 +23,17 @@ export default function connectNonCartesianSubplotToLayout(WrappedComponent) {
       this.fullContainer = fullContainer[subplot] || {};
 
       const trace = traceIndexes.length > 0 ? data[traceIndexes[0]] : {};
-      let fullTrace = {};
-      for (let i = 0; i < fullData.length; i++) {
-        if (traceIndexes[0] === fullData[i]._fullInput.index) {
-          /*
-           * Fit transforms are custom transforms in our custom plotly.js bundle,
-           * they are different from others as they create an extra trace in the
-           * data array. When plotly.js runs supplyTraceDefaults (before the
-           * transforms code executes) it stores the result in _fullInput,
-           * so that we have a reference to what the original, corrected input was.
-           * Then the transform code runs, our figure changes accordingly, but
-           * we're still able to use the original input as it's in _fullInput.
-           * This is the desired behaviour for our transforms usually,
-           * but it is not useful for fits, as the transform code adds some styles
-           * that are useful for the trace, so really for fits we'd like to read
-           * from _fullData, not _fullInput. Here we're setting _fullInput to
-           * _fullData as that is where the rest of our code expects to find its
-           * values.
-          */
-          if (trace.transforms && trace.transforms.every(t => t.type === 'fit')) {
-            fullData[i]._fullInput = fullData[i];
-          }
-
-          fullTrace = fullData[i]._fullInput;
-
-          break;
-        }
+      let fullTrace = fullData.filter(t => t && traceIndexes[0] === t.index)[0] || {};
```

Comment:
since this logic is in three places now, could we extract it out into a util or something?